src/backend/api/reservations.js

```diff
@@ -0,0 +1,98 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// let knexQuery = knex("reservations");
+
+//************ Returns all reservations ************
+
+router.get("/", async (request, response) => {
+  try {
+    // const reservations = await knex("reservations");
+    // response.json(reservations);
+
+    const checkIfAvailable = await knex("meals")
+      .join("reservations", "meals.id", "reservations.meal_id")
+      .select(
+        "reservations.meal_id",
+        "meals.id",
+        "meals.title",
+        "meals.max_reservations",
+        "reservations.number_of_guests"
+      )
+      .sum("reservations.number_of_guests")
+      .groupBy("meals.id");
+    // .where("meals.id", "reservations.meal_id");
+    response.json(checkIfAvailable);
+  } catch (error) {
+    throw error;
+  }
+});
+
+//************ Adds a new reservation ************
+router.post("/", async (request, response) => {
+  try {
+    // need to check if it is available to book
```

Comment:
Nice idea to check before inserting a new reservation!
However, it seems you're not doing any checks after all, but returning the result of your first query, which has the issues I mentioned above. If you want to keep this check, you should modify accordingly after you fix the query :)